index.d.ts

```diff
@@ -1,219 +1,278 @@
 export = Analytics;
+/**
+ * Represents a generic object in the APIs
+ * Use for parameters like context, traits etc.
+ */
+interface apiObject {
+  [index: string]:
+    | string
+    | number
+    | boolean
+    | apiObject
+    | (string | number | boolean | apiObject)[];
+}
+
+/**
+ * Represents the integration options object
+ * Example usages:
+ * integrationOptions { All: false, "Google Analytics": true, "Braze": true}
+ * integrationOptions { All: true, "Chartbeat": false, "Customer.io": false}
+ */
+interface integrationOptions {
+  // Defaults to true
+  All?: boolean;
+  // Destination name: true/false
+  [index: string]: boolean | undefined;
+}
+
+/**
+ * Represents the constructor options object
+ * Example usages:
+ * constructorOptions { flushAt: 20, "flushInterval": 20000, "enable": true, "maxInternalQueueSize":20000 }
+ */
+interface constructorOptions {
+  flushAt?: number | undefined;
+  flushInterval?: number | undefined;
+  enable?: boolean | undefined;
+  maxInternalQueueSize?: number | undefined;
```

Comment:
Why is `undefined` needed?
Also, same pattern followed in the rest of the definitions as well.

Reply:
Why is undefined needed? - Say the customer has implemented the constructor option object such a way that they need to send all the fields whether there is a value or not. To support this we need undefined. In case of undefined default values will be used.
Also, same pattern followed in the rest of the definitions as well - It was auto-generated by the compiler. So I didn't remove it.

Reply:
Are you saying with just `flushAt?: number` definition, `undefined` as argument value will not be acceptable?

Reply:
Yup, you are right `undefined` is not needed.